core/trino-main/src/main/java/io/trino/server/TaskResource.java

```diff
@@ -407,6 +409,16 @@ public void destroyTaskResults(
         asyncResponse.resume(Response.noContent().build());
     }
 
+    @ResourceSecurity(INTERNAL_ONLY)
+    @POST
+    @Path("pruneCatalogs")
+    @Consumes(MediaType.APPLICATION_JSON)
```

Comment:
I don't think `@Consumes` is necessary (most resources don't use it), and `@Produces` seems wrong since this is a void method.